otm/otm/entity/component.py

```diff
@@ -19,6 +23,30 @@ def __init__(self, component_id, name, component_type=None, parent=None, parent_
         self.threats: [ThreatInstance] = threats or []
         self.representations: List[RepresentationElement] = representations
 
+    @property
+    def id(self):
+        return self._id
+
+    @id.setter
+    def id(self, value):
+        self._id = truncate(value, MAX_ID_SIZE)
```

Comment:
Truncating the name seems good, but truncating the id might be inconsistent.
We have to think that the trustzone and component parents are  trustzone and component ids. The dataflow source and destination are component ids as well.
Is this change (the id truncated) needed to address the issue with the tags?
If is needed, we should truncate the tz and component  parent, and dataflow source and destination. If we do that, we need to assure that the relationships is done after truncation and we should add good tests to check this point.

Reply:
Good catch! I included the ID as well as a quick-win, but you're right that we can run into weird errors because of this. Probably, it's not so difficult to make the change consistent, but it's definitely out of the scope of this task and, as you mentioned, it should be properly tested. I'll remove the changes related to the ID.